src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -354,6 +386,123 @@ std::string RelationalExpression<Comparison>::asString(
+    AD_FAIL();
+  }
+};
+
+//______________________________________________________________________________
+template <IsDatatype Datatype>
+std::unique_ptr<PrefilterExpression> IsDatatypeExpression<Datatype>::clone()
+    const {
+  return std::make_unique<IsDatatypeExpression<Datatype>>(*this);
+};
+
+//______________________________________________________________________________
+template <IsDatatype Datatype>
+bool IsDatatypeExpression<Datatype>::operator==(
+    const PrefilterExpression& other) const {
+  const auto* otherIsDatatype =
+      dynamic_cast<const IsDatatypeExpression<Datatype>*>(&other);
+  if (!otherIsDatatype) {
+    return false;
+  }
+  // Given that the previous check doesn't fail and IsDatatypeExpression doesn't
+  // hold any member variables, the expressions are equal at this point.
+  return true;
+};
+
+//______________________________________________________________________________
+template <IsDatatype Datatype>
+std::string IsDatatypeExpression<Datatype>::asString(
+    [[maybe_unused]] size_t depth) const {
+  std::stringstream stream;
+  stream << "Prefilter IsDatatypeExpression:\n"
+         << "Prefilter for datatype: " << getDatatypeIsTypeStr(Datatype) << "."
+         << std::endl;
+  return stream.str();
+};
+
+//______________________________________________________________________________
+template <IsDatatype Datatype>
+std::vector<BlockMetadata> IsDatatypeExpression<Datatype>::evaluateImpl(
+    std::span<const BlockMetadata> input, size_t evaluationColumn) const {
+  using enum IsDatatype;
+
+  // Vector for storing bounding ValueIds to the BlockMetadata values.
+  std::vector<ValueId> valueIdsInput;
+  valueIdsInput.reserve(2 * input.size());
+
+  std::ranges::for_each(
+      input, [&valueIdsInput, evaluationColumn](const BlockMetadata& block) {
+        valueIdsInput.push_back(
+            getIdFromColumnIndex(block.firstTriple_, evaluationColumn));
+        valueIdsInput.push_back(
+            getIdFromColumnIndex(block.lastTriple_, evaluationColumn));
+      });
+
+  auto idsBegin = valueIdsInput.begin();
+  auto idsEnd = valueIdsInput.end();
+  std::vector<RandomItPair> relevantRanges;
+  if constexpr (Datatype == BLANK) {
+    relevantRanges.push_back(valueIdComparators::getRangeForDatatype(
+        idsBegin, idsEnd, Datatype::BlankNodeIndex));
+  } else if (Datatype == NUMERIC) {
+    relevantRanges.push_back({valueIdComparators::getRangeForDatatype(
+                                  idsBegin, idsEnd, Datatype::Int)
+                                  .first,
+                              valueIdComparators::getRangeForDatatype(
+                                  idsBegin, idsEnd, Datatype::Double)
+                                  .second});
+  } else {
+    assert(Datatype == IRI || Datatype == LITERAL);
+    // Not implemented yet, i assume the prefiltering can be even more precise
+    // by making use of `transformToFirstPossibleBiggerValue`.
```

Comment:
We were already wondering, why those expressions weren't working. There are Several ways to do this, but yes, maybe it is related to the prefix filtering. Maybe we can discuss how far you are with that one.
For now, you should put a hard failure here, and not add these prefilters when planning. That way we can keep the infrastructure in this PR, and still not have wrong query results.